### ISSUE TITLE
SASL protocol error

### ISSUE THREAD

**Issue by AndrioCelos** [kvirc/KVIrc#1677]
KVIrc is sending an extra null byte after the password when authenticating via SASL PLAIN:

```
[18:56:33] AUTHENTICATE ***redacted***
[18:56:34] :local.test.redatcted 904 061AAAAAW :SASL authentication failed
```

This is not in accordance with [RFC 4616](https://tools.ietf.org/html/rfc4616), and causes authentication to fail with some services (Anope).

**Comment by ctrlaltca** [comment]
I've redacted your comment containing a base64-encoded password.

**Comment by un1versal** [comment]
Confirmed fixed by #1680
